src/ConnectedMode/Migration/ConnectedModeMigration.cs

```diff
@@ -42,19 +44,53 @@ private sealed class ChangedFiles : List<FilePathAndContent<string>> { }
         private readonly IFileProvider fileProvider;
         private readonly IFileCleaner fileCleaner;
         private readonly IVsAwareFileSystem fileSystem;
+        private readonly ISonarQubeService sonarQubeService;
         private readonly ILogger logger;
 
+        // The user can have both the legacy and new connected mode files. In this case the behaviour is slightly different.
```

Comment:
`In this case the behaviour is slightly different.` isn't very enlightening ;-)

Maybe something like `In that case, we expect the SonarQubeService to already be connected.` ?